### ISSUE TITLE
Individual mrc files (3D volumes) not showing up in data page

### ISSUE THREAD

**Issue by mcianfrocco** [cianfrocco-lab/COSMIC-CryoEM-Gateway#160]
Hi @monaw - I just tried to upload a single .mrc file via Globus and the file is not being displayed in the data page. Previously this worked, so I'm wondering if the folder naming issues are related to this.

**Comment by monaw** [comment]
ah, for Globus transfers, the code only lists *.star files, not *.mrc.  If you upload using the browser method, then *.mrc will be listed in the data folder.

**Comment by mcianfrocco** [comment]
This wasn't a problem previously - maybe something changed?

**Comment by monaw** [comment]
@mcianfrocco - do you think user will be uploading this file separately or together with their other data (i.e. star and image files)?

**Comment by mcianfrocco** [comment]
They would upload this separately

**Comment by monaw** [comment]
@mcianfrocco - i just did a test and it worked, using Globus to upload a single .mrc file.  can you test again and let me know?  if it doesn't work, is your mrc file in a directory?

**Comment by mcianfrocco** [comment]
@monaw  - I just tried this today and I an error: 

May 28 2019 10:52:54 [xec-24] ERROR - ceptionMappingInterceptor:42     | /projects/cosmic2/credential/user_cred.pem (No such file or directory)
java.io.FileNotFoundException: /projects/cosmic2/credential/user_cred.pem (No such file or directory)

It seems that I can't get any globus uploads now. @kenneth59715 can you also try? Maybe a credential problem?

**Comment by kenneth59715** [comment]
I was able to upload through browser, no problem.

The globus upload glitched with an error reported for
no directory, but eventually succeeded.  Does it still
fail?

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by mcianfrocco** [comment]
No - it doesn't go through. It also doesn't appear in the transfers history. I just tried again and it didn't work. 

I also created a new directory with a unique name that I know I haven't used before (in case that was the problem) and I still had an error.

**Comment by monaw** [comment]
@mcianfrocco - ok, I've put the /projects/cosmic2/credential/ directory back (oddly it worked fine for transfer without it when I tested after the new storage change).  please test again and let me know if you are still getting that error.  i just did a test and it worked for me.

**Comment by monaw** [comment]
@kenneth59715 - can you do a test upload of mrc file using Globus?

**Comment by mcianfrocco** [comment]
I just got the same upload error on the gateway

**Comment by mcianfrocco** [comment]
Error: 

May 28 2019 19:42:44 [xec-18] ERROR - ceptionMappingInterceptor:42     | Cannot run program "/projects/cosmic2/mkproxy/mkproxy": error=2, No such file or directory

**Comment by monaw** [comment]
ok, i saw that too and have put those files back also; please try one more time when you get a chance...

**Comment by mcianfrocco** [comment]
Another error: 

<img width="1151" alt="Screen Shot 2019-05-28 at 11 37 44 PM" src="https://user-images.githubusercontent.com/1464469/58527687-ae918180-81a1-11e9-8366-6616a03072f4.png">

**Comment by mcianfrocco** [comment]
So - the transfer showed up as successful on the gateway, but the file did not get moved onto comet. I don't see any files present on the gateway

**Comment by mcianfrocco** [comment]
I should add that this is related to uploading a single .mrc file

**Comment by mcianfrocco** [comment]
One more update - I was able to successfully upload a directory with a .star file. The files were transferred and showed up into the gateway data page correctly. 

So - @monaw - when you transferred the single file, did you check if it was actually moved onto SDSC?

**Comment by kenneth59715** [comment]
Scratch the confirmation.  The single file does get uploaded
to db_documents.  Browser upload does not end up in globus_transfers,
obviously.  Michael, are you doing single file uploads with globus upload?

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by kenneth59715** [comment]
I tried a single file globus upload.  It does not show up in globus_transfers
on Comet, but does appear in my Data List.  I don't see errors in
the logs.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by monaw** [comment]
A single file upload is handled through CIPRES the way it did originally, regardless through browser or Globus.  This means if done through Globus, the file will initially show up in the globus_transfers/ directory but then will be moved to db_documents/... once "Refresh" is done.  If uploaded through browser, it does NOT show up in globus_transfers/ and is directly placed in db_documents/... Either way, the single file should show up in the gateway UI.

globus_transfers/ ONLY contain directory data.

@mcianfrocco - I saw the same error you showed above when I did my first transfer but then didn't see it again subsequently.  Upon some debugging, it seems to appear the first time a user does a Globus transfer and the runtime error is 

```
[...]
May 28 2019 20:37:29 [xec-14] ERROR -            TransferAction:895    |        Create directory: EndpointError(502 Bad Gateway) on request '62Gm5jkgN'         resource '/endpoint/de463f97-6d04-11e5-ba46-22000b92c6ec/mkdir': Endpoint       Error, Error (mkdir)
 Endpoint: XSEDE Comet (de463f97-6d04-11e5-ba46-22000b92c6ec)
 Server: oasis-dm.sdsc.xsede.org:2811
 Command: MKD /projects/cosmic2/gateway/globus_transfers/mcianfro@umich.edu/     testing20190528/
 Message: Fatal FTP Response
 ---
 Details: 550-GlobusError: v=1 c=PATH_NOT_FOUND\r\n550-GridFTP-Errno: 2\r\n550-  GridFTP-Reason: System error in mkdir\r\n550-GridFTP-Error-String: No such      file or directory\r\n550 End.\r\n
 
 May 28 2019 20:37:29 [xec-14] INFO  -               NgbwSupport:1009   |        mcianfro@umich.edu - Error, unable to access XSEDE Comet storage.
[...]
```

More debugging is needed to figure out why it fails first time...unfortunately looks like this WILL show for every user once.   Might want to warn our beta testers...

I suggest we open a new ticket for this, assign it to me and I'll work on it in dev.  Or if you want to hold off until this is resolved in production...my estimate is that it shouldn't take long...maybe a few hours to a day but i won't be able to work on this until my next project time...next thursday.

**Comment by mcianfrocco** [comment]
@kenneth59715 - I have done both web and globus upload for file upload. The reason for using globus to upload single files is that sometimes the 3D volumes are sitting on my server with my data and I'm using the gateway from my laptop.

I will make a note that users must upload 3D volumes through the web interface for now. 

@monaw - I'll open a new issue related the first time transfer. 

I will keep this open as I think it will be good to have the flexibility of uploading single files through globus or the web.

**Comment by kenneth59715** [comment]
If I understand Mona's explanation, both globus upload and browser
upload for single files should work.  You won't see the file in
globus_transfers, though.  It should get put into the job working
directory when the job is started.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by mcianfrocco** [comment]
Oh right - yes, I see. Thanks @kenneth59715 for clarifying. 

I also just double checked and this worked. I am closing this issue now : )
